compiler/rustc_codegen_llvm/src/intrinsic.rs

```diff
@@ -1377,31 +1380,25 @@ fn generic_simd_intrinsic<'ll, 'tcx>(
 
         // The element type of the third argument must be a signed integer type of any width:
         let (_, element_ty2) = arg_tys[2].simd_size_and_type(bx.tcx());
-        match element_ty2.kind() {
-            ty::Int(_) => (),
+        let mask_elem_bitwidth = match element_ty2.kind() {
```

Comment:
This changes the `simd_gather` intrinsic, but none of the tests tests that were touched in this test involve it? Do we have no `simd_gather` tests? Please add one or adjust an existing one if there is one.

Reply:
We do have existing `simd_gather` and `simd_scatter` tests, but I believe none of the existing tests are affected by this change. Is that a problem, exactly?

Reply:
I would say that this exposes a hole/deficiency in the test suite coverage. Ideally we take care of it at the same time we find out about this, but I’d be okay with an issue describing tests that should be added or even just not doing anything if y'all are comfortable with not having test coverage here.

That said, seeing test changes would definitely have helped with reviewing this PR.

Reply:
I just realized I still had some uncommitted changes to the tests locally :( will push those in a minute.

There were codegen tests for gather/scatter, but they only checked that the llvm intrinsic was correctly emitted and did not verify how the mask was converted to `i1`. I adjusted these tests.

What might be missing is more assembly tests for these intrinsics that verify no unnecessary shifts or other instructions get emitted.